### PR TITLE
remove the tildot transform

### LINKED ISSUE #2951: remove tildot support
## What is the Problem Being Solved?

In #2503 I think we came to a consensus that we should remove support for tildot (`target~.method(args)`), and write `E(target).method(args)` instead. The parser is hard to manager.

If we indeed have consensus on that, the tasks are:

* change `bundleSource` to remove the tildot transform we apply during `rollup()`
* in swingset, remove `vatPowers.transformTildot`
  * in `controller.js` where it provides `transformTildot` in `kernelEndowments`
  * this changes `kernel.js` where it builds `allVatPowers`
  * and also `liveSlots.js` where it builds `baseVP`

We'll also remove a couple of test cases.

@erights @Chris-Hibbert I think you two were driving the decision-making process on #2503, should I proceed to remove tildot support now?

### PR DESCRIPTION
**Pull request by warner**
This removes the tildot transform from `bundleSource()` and from swingset's `vatPowers.transformTildot`, as well as all the code (examples and tests) which was using it.

closes #2951

### CODE REVIEW

**Comment by warner** [comment]
@michaelfig I'd like your eyes on the `bundle-source` changes, @dckc on the vat-manager changes.

**Review by michaelfig — APPROVED** [review]
LGTM.

Farewell, my only friend. 😆

**Comment by michaelfig** [comment]
Oh, BTW, `fix!(bundle-source):` should be `fix(bundle-source)!:`. same with the other breaking fix.

File: packages/SwingSet/docs/liveslots.md [hunk 1]
```diff
@@ -4,8 +4,6 @@ This runtime treats vats as if they were userspace processes in an operating sys
 
 "Liveslots" is a particular dispatch mechanism that uses Maps/WeakMaps to track "live objects". Imported references to objects on other vats are represented by special "Presence" objects. Liveslots implements the `E` wrapper which allows messages to be sent through presences: `promise = E(presence).methodname(args)`. Presences can be sent in arguments or returned from method calls.
 
-In the future, we expect to have a syntax desugaring phase, so `E(p).foo(arg)` can be written as `p~.foo(arg)`. This "tildot notation" indicates that we're effectively calling `p.foo(arg)`, but 1: it always happens in a later turn (and always returns a Promise), and 2: `p` can refer to something on a remote host.
```
Comment (dckc) [inline packages/SwingSet/docs/liveslots.md:7]: We don't even expect to have this in the future?

**Review by dckc — APPROVED** [review]
looks good.

here's hoping for when we can get back to this...

File: packages/SwingSet/docs/liveslots.md (hunk 1 above)
Comment (warner) [reply packages/SwingSet/docs/liveslots.md:7]: I think we *are* expecting to bring tildot back some day, but it didn't seem useful to mention that hope in these docs, nor to explain the eventual eventual-send syntax. I think I left one note about future plans somewhere (now I forget where), and that seemed like enough.

**Pull request merged**